plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/util/SynthesizedColumnHandler.java

```diff
@@ -95,7 +108,7 @@ private void initPartitionKeyStrategies(ImmutableMap.Builder<String, Synthesized
     {
         for (HivePartitionKey partitionKey : hudiSplit.getPartitionKeys()) {
             builder.put(partitionKey.name(), (blockBuilder, type) ->
-                    HudiAvroSerializer.appendTo(type, partitionKey.value(), blockBuilder));
+                    appendToBuilder(type, partitionKey.value(), blockBuilder));
```

Comment:
Does this affect partition pruning or it only affects the values in the synthesized columns?

Reply:
Does not affect partition pruning. It only affects the values in synthesized columns. 
Partition columns does not involve SynthesizedColumnHandler. The handler is only used to build blocks when scanning is actually done. i.e. Not involved in planning stage.